### PR TITLE
benchmark_app: except ALLOW_AUTO_BATCHING

### PR DESCRIPTION
**Pull request by Wovchena**
Running benchmark_app.py with -b 1 -d CPU fails with `Unsupported property ALLOW_AUTO_BATCHING by CPU plugin`.

C++ benchmark_app sets ALLOW_AUTO_BATCHING in
ov::Core::compile_model() call, which doesn't trigger the error.

ticket: _CVS-112017_

### CODE REVIEW

File: tools/benchmark_tool/openvino/tools/benchmark/benchmark.py [hunk 1]
```diff
@@ -60,7 +60,12 @@ def set_cache_dir(self, cache_dir: str):
         self.core.set_property({'CACHE_DIR': cache_dir})
 
     def set_allow_auto_batching(self, flag: bool):
-        self.core.set_property({'ALLOW_AUTO_BATCHING': flag})
+        try:
+            self.core.set_property({'ALLOW_AUTO_BATCHING': flag})
```
Comment (nkogteva) [inline tools/benchmark_tool/openvino/tools/benchmark/benchmark.py:64]: In C++ this flag was moved to device config. We should align it https://github.com/openvinotoolkit/openvino/blob/71dcdf8a28b0c432e835e5ec8d52bf34fa07a7a6/samples/cpp/benchmark_app/main.cpp#L553

Comment (Wovchena) [reply tools/benchmark_tool/openvino/tools/benchmark/benchmark.py]: Aligned

**Review by WeldonWangwang — APPROVED** [review]

**Review by ilya-lavrenov — APPROVED** [review]

**Review by songbell — APPROVED** [review]
@peterchen-intel @WeldonWangwang  we need to enable related tests with python app
@ilya-lavrenov  shall we enable tests in precommit as well?

**Review by songbell — APPROVED** [review]
@peterchen-intel @WeldonWangwang  we need to enable related tests with python app
@ilya-lavrenov  shall we enable tests in precommit as well?

**Comment by ilya-lavrenov** [comment]
> @peterchen-intel @WeldonWangwang we need to enable related tests with python app @ilya-lavrenov shall we enable tests in precommit as well?

yes, we have samples tests in precommit. It would be great to extend them to more scenarios

**Pull request merged**